sources/academy/webscraping/scraping_basics_python/05_parsing_html.md

```diff
@@ -0,0 +1,211 @@
+---
+title: Parsing HTML with Python
+sidebar_label: Parsing HTML
+description: Lesson about building a Python application for watching prices. Using the Beautiful Soup library to parse HTML code of a product listing page.
+sidebar_position: 5
+slug: /scraping-basics-python/parsing-html
+---
+
+**In this lesson we'll look for products in the downloaded HTML. We'll use BeautifulSoup to turn the HTML into objects which we can work with in our Python program.**
+
+---
+
+From lessons about browser DevTools we know that the HTML tags representing individual products have a `class` attribute which, among other values, contains `product-item`.
+
+![Products have the ‘product-item’ class](./images/product-item.png)
+
+As a first step, let's try counting how many products are on the listing page.
+
+## Treating HTML as a string
+
+Currently, the entire HTML is available in our program as a string. Our program can print it to the screen or save it to a file, but not much more. If it's a string, could we use Python string operations to count the products? Each Python string has `.count()`, a [method for counting substrings](https://docs.python.org/3/library/stdtypes.html#str.count).
+
+After manually inspecting the page in browser DevTools we can see that all product cards have the following structure:
+
+```html
+<div class="product-item product-item--vertical ...">
+  <a href="/products/..." class="product-item__image-wrapper">
+    ...
+  </a>
+  <div class="product-item__info">
+    ...
+  </div>
+</div>
+```
+
+At first sight, counting `product-item` occurrences wouldn't match only products, but also `product-item__image-wrapper`. Hmm.
+
+We could try looking for `<div class="product-item`, a substring which represents the enitre beginning of each product tag, but that would also count `<div class="product-item__info`! We'll need to add a space after the class name to avoid matching those. Replace your program with the following code:
+
+```py
+import httpx
+
+url = "https://warehouse-theme-metal.myshopify.com/collections/sales"
+response = httpx.get(url)
+response.raise_for_status()
+
+html_code = response.text
+count = html_code.count('<div class="product-item ')
+print(count)
+```
+
+Note that because the substring contains a double quote character, we need single quotes as string boundaries.
```

Comment:
maybe move that part as a comment within code block? it kind of feels weird to have a note that is not an admonition and then another admonition below it

Reply:
Addressed in 9ed1f738